notification-service/src/constants/SUBSCRIPTIONS.ts

```diff
@@ -0,0 +1,52 @@
+import axios from 'axios';
+import { EventSubscription } from '../types';
+import { ETHERSCAN_API_KEY, ETHEREUM_NETWORK } from '../variables';
+import { getEtherscanAPI, getEtherscanURL } from './NETWORKS';
+import { ethers } from 'ethers';
+
+export async function getAbiFromContractAddress(contract: string) {
+    if (!ETHERSCAN_API_KEY) {
+        throw new Error('subscriptions: please set a valid ETHERSCAN_API_KEY, in the environment files');
+    }
+    const result = await axios.get(
+        `${getEtherscanAPI(
+            ETHEREUM_NETWORK
+        )}/api?module=contract&action=getabi&address=${contract}&apikey=${ETHERSCAN_API_KEY}`
+    );
+    return JSON.parse(result.data.result);
+}
+
+export const SUBSCRIPTIONS: EventSubscription[] = [
+    {
+        id: 'ChainLogUpdateAddress',
+        contract: '0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F',
+        eventName: 'UpdateAddress',
+        formatData: event => {
+            return `
+                Key: ${ethers.utils.formatBytes32String(event.args.key)}<br /> 
+                Address: <a href="${getEtherscanURL(ETHEREUM_NETWORK)}/address/${
+                event.args.addr
+            }" target="_blank" style="color: lightblue;">${event.args.addr}</a>
+            `;
+        },
```

Comment:
If you're using `getEtherscanURL` in every event, you can nicely provide it as a helper via function argument (with preset NETWORK) `getEtherscanURL(event.args.addr, 'address')`

```suggestion
        formatData: ({ event, getEtherscanURL }) => {
            const contractName = ethers.utils.formatBytes32String(event.args.key);
            const url = getEtherscanURL(event.args.addr, 'address')
            return `
                Contract: ${contractName}<br /> 
                Address: <a href="${url}" target="_blank" style="color: lightblue;">${event.args.addr}</a>
            `;
        },
```